### PR TITLE
[Serde generate] improve python codegen

### PR DESCRIPTION
**Pull request by matbd**
## Summary

* Fix formatting of comments
* omit INDEX and VARIANTS when serialization is skipped
* (possibly breaking) set `# pyre-strict` in generated code

## Test Plan

include unit tests + test in Libra

### CODE REVIEW

File: serde-generate/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "serde-generate"
-version = "0.10.0"
+version = "0.11.0"
```
Comment (bmwill) [inline serde-generate/Cargo.toml:3]: This doesn't appear to be a breaking API change? maybe you can just do a point release?

Comment (matbd) [reply serde-generate/Cargo.toml:3]: I wonder if adding `#pyre-strict` could trigger a chain reaction with more typing errors in python user code.

Comment (matbd) [reply serde-generate/Cargo.toml:3]: I guess it depends if `breaking` is defined in terms of Rust compilation and behaviors only :)

Comment (bmwill) [reply serde-generate/Cargo.toml:3]: eh fair point :) I was strictly thinking in terms of the crate's API but i suppose "breaking" has other  implications as well.

**Pull request merged**